cognite/client/data_classes/data_modeling/instances.py

```diff
@@ -561,6 +560,62 @@ class InstanceAggregationResultList(CogniteResourceList[InstanceAggregationResul
     _RESOURCE = InstanceAggregationResult
 
 
+class InspectionResults(DataModelingResource):
+    def __init__(self, involved_views: list[ViewId], involved_containers: list[ContainerId]) -> None:
+        self.involved_views = involved_views
+        self.involved_containers = involved_containers
+
+    @classmethod
+    def _load(cls, resource: dict, cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            involved_views=[ViewId.load(vid) for vid in resource["involvedViews"]],
+            involved_containers=[ContainerId.load(cid) for cid in resource["involvedContainers"]],
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        views = [vid.dump(camel_case, include_type=True) for vid in self.involved_views]
+        containers = [cid.dump(camel_case, include_type=True) for cid in self.involved_containers]
+        return {
+            "involvedViews" if camel_case else "involved_views": views,
+            "involvedContainers" if camel_case else "involved_containers": containers,
+        }
+
+
+class InstanceInspectResult(DataModelingResource):
+    def __init__(
+        self,
+        space: str,
+        external_id: str,
+        instance_type: Literal["node", "edge"],
+        inspection_results: InspectionResults,
+    ) -> None:
+        self.space = space
+        self.external_id = external_id
+        self.instance_type = instance_type
+        self.inspection_results = inspection_results
+
+    @classmethod
+    def _load(cls, resource: dict, cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            space=resource["space"],
+            external_id=resource["externalId"],
+            instance_type=resource["instanceType"],
+            inspection_results=InspectionResults.load(resource["inspectionResults"]),
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {
+            "space": self.space,
+            "externalId" if camel_case else "external_id": self.external_id,
+            "instanceType" if camel_case else "instance_type": self.instance_type,
+            "inspectionResults" if camel_case else "inspection_results": self.inspection_results.dump(camel_case),
+        }
+
+
+class InstanceInspectResultList(CogniteResourceList[InstanceInspectResult]):
```

Comment:
Since we rely on ordering, I think this result class should mirror how it's done for /byids (`.retrieve()`), where we have `nodes` and `edges` attributes.